android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressIndicator.kt

```diff
@@ -81,22 +81,23 @@ fun PagerProgressIndicator(
         .coerceAtMost(pagerState.pageCount)
         .toOddUnder()
 
-    val indicatorWidth = selectedItemSize * adjustedMaxItems +
+    val rowWidth = selectedItemSize * adjustedMaxItems +
         indicatorSpacing * (adjustedMaxItems - 1)
 
-    val widthInPx = with(LocalDensity.current) { selectedItemSize.toPx() }
+    val itemWidthInPx = with(LocalDensity.current) { selectedItemSize.toPx() }
+    val rowWidthInPx = with(LocalDensity.current) { rowWidth.toPx() }
     LaunchedEffect(pagerState.currentPage) {
-        val viewportSize = listState.layoutInfo.viewportSize
-
         listState.animateScrollToItem(
             index = pagerState.currentPage,
-            scrollOffset = (widthInPx / 2 - viewportSize.width / 2).toInt(),
+            scrollOffset = (itemWidthInPx / 2 - rowWidthInPx / 2).toInt()
         )
     }
 
+    val numberOfScaledItems = numberOfItemsToScale.coerceAtMost(adjustedMaxItems).toOddUnder()
```

Comment:
Key change 3: Ensure that the number of scaled items is capped to the number of total items displayed. E.g. can't scale the dot size over 5 items if only 3 dots are displayed.